inginious/frontend/static/js/task_dispensers.js

```diff
@@ -382,6 +388,32 @@ function dispenser_util_get_section_config(element) {
     return config_list;
 }
 
+function dispenser_util_get_weights() {
+    const weight_list = {};
+    $(".weight").each(function(index){
+        weight = parseFloat(this.value)
+        if(weight > 1){
+            weight_list[this.id] = weight
+        }
+        else if(isNaN(weight) && this.value !== ""){
```

Comment:
Those error messages should better be generated from the Python code, as it must be ensured that anarbitrary POSTed  form will get the same treatment and not inject bad values in database/files. I guess these checks can be moved to the ``check_dispenser_data`` instead.

Furthermore, all positive values can be (and were) tolerated, so values between 0 and 1 should be accepted. I guess one may have the practice to use several weight values that sum to 1.